.gitlab-ci.yml

```diff
@@ -1,20 +1,116 @@
-######################################################################################
-# Copyright 2020 Lawrence Livermore National Security, LLC and other CARE developers.
-# See the top-level LICENSE file for details.
+###############################################################################
+# Copyright 2021 Lawrence Livermore National Security, LLC and other CARE
+# developers.
 #
 # SPDX-License-Identifier: BSD-3-Clause
-######################################################################################
+#
+# Based on implementation in the Umpire Project,  Copyright (c) 2018-20, 
+# Lawrence Livermore National Security, LLC and Umpire project contributors.
+#
+# See the COPYRIGHT file for details.
+###############################################################################
+
+###############################################################################
+# General GitLab pipelines configurations for supercomputers and Linux clusters
+# at Lawrence Livermore National Laboratory (LLNL).
+#
+# This entire pipeline is LLNL-specific
+# #############################################################################
+
+# We define the following GitLab pipeline variables:
+#
+# GIT_SUBMODULE_STRATEGY:
+# Tells Gitlab to recursively update the submodules when cloning umpire
+#
+# ALLOC_NAME:
+# On LLNL's ruby, this pipeline creates only one allocation shared among jobs
+# in order to save time and resources. This allocation has to be uniquely named
+# so that we are sure to retrieve it.
+#
+# BUILD_ROOT:
+# The path to the shared resources between all jobs. The BUILD_ROOT is unique to
+# the pipeline, preventing any form of concurrency with other pipelines. This
+# also means that the BUILD_ROOT directory will never be cleaned.
+# TODO: add a clean-up mechanism
 
 variables:
-   GIT_SUBMODULE_STRATEGY: normal
+  GIT_SUBMODULE_STRATEGY: recursive
```

Comment:
Add a TODO to change this from `recursive` to `normal`. I tried to flatten all the dependencies in CARE, but I'm missing a hip/rocm library that RAJA depends on, I think.